File: mybudget-backend/src/main/java/com/Project/MyBudget/services/expense/ExpenseService.java
package com.Project.MyBudget.services.expense;

import com.Project.MyBudget.dto.ExpenseDTO;
import com.Project.MyBudget.entity.Expense;

import java.util.List;

public interface ExpenseService {

    Expense postExpense(ExpenseDTO expenseDTO);

    List<Expense> getAllExpenses();

    Expense getExpenseById(Long id);

    Expense updateExpense(Long id, ExpenseDTO expenseDTO);

    void deleteExpenseById(Long id);
}

File: mybudget-frontend/src/app/services/expense/expense.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ExpenseService {

  private baseUrl = `${environment.apiUrl}api/expense`;

  constructor(private http: HttpClient) { }

  postExpense(expenseDTO: any): Observable<any> { 
    return this.http.post(this.baseUrl, expenseDTO);
  }

  getAllExpenses(): Observable<any> { 
    return this.http.get(`${this.baseUrl}/all`);
  }

  deleteExpense(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`);
  }

  getExpenseById(id: number): Observable<any> { 
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  updateExpense(id: number, expenseDTO: any): Observable<any> { 
    return this.http.put(`${this.baseUrl}/${id}`, expenseDTO);
  }
}

